lib/rate_limiting.rb

```diff
@@ -176,6 +183,8 @@ def allowed?(request)
       return true if whitelist?(request.ip)
       return false if blacklisting_ip(request)
       if rule = find_matching_rule(request)
+        xff_ip = rule.get_xff_ip(request)
+        return true if xff_ip.present? && whitelist?(xff_ip, true)
```

Comment:
instead of changing the signature of whitelist?(), can we do below change
```
xff_ip = rule.get_xff_ip(request)
return true if xff_ip.present? && (@whitelist = nil || whitelist?(xff_ip))
```

Reply:
Sure done